tools/trap/ramdumpParser.py

```diff
@@ -620,39 +626,63 @@ def find_crash_point(log_file):
-        # If yes, print the crash point using addr2line
-        f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(lr_value))
-        result = f.read()
-        if '??' not in result and '$d' not in result:
-            print('Kernel Crash point is as follows:')
-            print('[ Caller - return address (LR) - of the function which has caused the crash ]')
-            format_output(result)
-    elif (address2 >= hex(g_stext_flash) and address2 < hex(g_etext_flash)) or (address2 >= hex(g_stext_ram) and address2 < hex(g_etext_ram)):
-        f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(pc_value))
-        result = f.read()
-        if '??' not in result and '$d' not in result:
-            print('Kernel Crash point is as follows:')
-            print('[ Current location (PC) of assert ]')
-            print(' - Exact crash point might be -4 or -8 bytes from the PC.')
-            format_output(result)
-    elif (not is_app_crash):
-        print('PC & LR values not in any text range, no probable crash point detected')
-        print('-----------------------------------------------------------------------------------------\n')
+    # Check for pc & lr values in application text address range
+    if (not is_app_crash):
+        # Check for pc & lr values in kernel text address range
+        if (address1 >= hex(g_stext_flash) and address1 < hex(g_etext_flash)) or (address1 >= hex(g_stext_ram) and address1 < hex(g_etext_ram)):
+            # If yes, print the crash point using addr2line
+            f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(lr_value))
+            result = f.read()
+            if '??' not in result and '$d' not in result:
+                is_kernel_crash = 1
+                print('Kernel Crash point is as follows:')
+                print('[ Caller - return address (LR) - of the function which has caused the crash ]\n')
+                format_output(result)
+        if (address2 >= hex(g_stext_flash) and address2 < hex(g_etext_flash)) or (address2 >= hex(g_stext_ram) and address2 < hex(g_etext_ram)):
+            f = os.popen('arm-none-eabi-addr2line -a -f -e' + elf_path + ' ' + hex(pc_value))
+            result = f.read()
+            if '??' not in result and '$d' not in result:
+                is_kernel_crash = 1
+                print('\nKernel Crash point is as follows:')
+                print('[ Current location (PC) of assert ]')
+                print(' - Exact crash point might be -4 or -8 bytes from the PC.\n')
+                format_output(result)
+
+        #It displays the debug symbols corresponding to all the addresses in the kernel text address range
+        print('\n--------------------------- DEBUG SYMBOLS IN KERNEL TEXT RANGE --------------------------')
+        os.system("python ../debug/debugsymbolviewer.py System.map log_file 0x0 1")
```

Comment:
Could you check if kernel symbols are being printed twice? cause line 680-681 has the same code.

Reply:
yes, they are being printed twice.